repos/spack_repo/builtin/packages/ambertools/package.py

```diff
@@ -44,7 +56,8 @@ class Ambertools(CMakePackage):
         type=("build", "run"),
     )
     # Python dependencies
-    depends_on("python@3.8:3.10 +tkinter", type=("build", "run"))
+    depends_on("python@3.8:3.10 +tkinter", when="^22jlmrcc", type=("build", "run"))
+    depends_on("python@3.11:", when="^25", type=("build", "run"))
```

Comment:
the `when` conditions are wrong. e.g., it's `@25:` which means version 25 or newer.